crypto/engine/eng_devcrypto.c

```diff
@@ -327,10 +396,12 @@ static void prepare_cipher_methods(void)
             cipher_driver_info[i].status = DEVCRYPTO_STATUS_NO_CIOCGSESSION;
             continue;
         }
+        cipher_mode = cipher_data[i].flags & EVP_CIPH_MODE;
 
         if ((known_cipher_methods[i] =
                  EVP_CIPHER_meth_new(cipher_data[i].nid,
-                                     cipher_data[i].blocksize,
+                                     cipher_mode == EVP_CIPH_CTR_MODE ? 1 :
+                                                    cipher_data[i].blocksize,
```

Comment:
Isn't it easier to change the block size for CTR mode ciphers in `cipher_data`?

Reply:
The CTR mode ciphers behave like (many will say the are) stream ciphers, but they are handled in blocks.  So from openssl looking at the engine point of view, block size is 1, but internally, it has to be aware of the block size to do its job--for patial blocks, a full block is calculated, then as data arrives, it gets xored with the result.

Reply:
Ah, good point.